azure-mgmt-containerregistry/src/main/java/com/microsoft/azure/management/containerregistry/Webhook.java

```diff
@@ -0,0 +1,663 @@
+
+        /**
+         * The stage of the webhook definition allowing to specify the custom headers that will be added to the notifications.
+         *
+         * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+         */
+        interface WithCustomHeaders<ParentT> {
+            /**
+             * Specifies custom headers that will be added to the notifications.
+             *
+             * @param name of the optional header
+             * @param value of the optional header
+             * @return the next stage of the definition
+             */
+            WithAttach<ParentT> withCustomHeader(String name, String value);
+
+            /**
+             * Specifies the the service URI for post notifications.
+             *
+             * @param customHeaders the "Name=Value" custom headers.
+             * @return the next stage of the definition
+             */
+            WithAttach<ParentT> withCustomHeaders(Map<String, String> customHeaders);
+        }
+
+        /**
+         * The stage of the webhook definition allowing to specify the scope of repositories where the event can be triggered.
+         *
+         * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+         */
+        interface WithRepositoriesScope<ParentT> {
+            /**
+             * Specifies the scope of repositories where the event can be triggered.
+             * <p>
+             * For example, 'foo:*' means events for all tags under repository 'foo'. 'foo:bar' means events for 'foo:bar' only.
+             *  'foo' is equivalent to 'foo:latest', empty means all events.
+             *
+             * @param repositoriesScope the scope of repositories where the event can be triggered; empty means all events
+             * @return the next stage of the definition
+             */
+            WithAttach<ParentT> withRepositoriesScope(String repositoriesScope);
+        }
+
+        /**
+         * The stage of the webhook definition allowing to specify the default status of the webhook after being created.
+         *
+         * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+         */
+        interface WithDefaultStatus<ParentT> {
+            /**
+             * Specifies the default status of the webhook; default is "enabled".
+             *
+             * @param defaultStatus indicates whether the webhook is enabled or disabled after the being created
+             * @return the next stage of the definition
+             */
+            WithAttach<ParentT> withDefaultStatus(WebhookStatus defaultStatus);
+        }
+
+        /**
+         * The stage of the webhook definition allowing to specify the tags.
+         *
+         * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+         */
+        interface WithTags<ParentT> {
```

Comment:
have you considered referencing `Resource.DefinitionWithTags<ReturnT>` instead of defining this explicitly here?

Reply:
yes but could not work around the "clashing" return types; there are couple definitions for these with'ers that result in errors when trying to merge them all into one Impl.

Reply:
There is `Resource.DefinitionWithTags` for reference from definitions, and `Resource.UpdateWithTags` for reference from Updates. Since this is possibly the first time we're exposing tags on this type of child, what we could add is `Resource.UpdateDefinitionWithTags` for that 3rd scenario (parent update/child definition)

Reply:
the issue is the Resource.DefinitionWithTags and Resource.UpdateDefinitionWithTags can only be extended one time.